src/main/java/org/jenkinsci/plugins/ccxml/CCXMLAction.java

```diff
@@ -51,14 +62,45 @@ public String getDisplayName() {
 
     @Override
     public String getUrlName() {
-        return "cc.xml2";
+        return URL_NAME;
     }
 
     public View getView() {
         return this.view;
     }
 
     /**
+     * @return A map containing the items in the view object. If the request
+     * contains a query parameter named "recursive", then folders in the view
+     * are traversed recursively and all items in those folders are returned
+     * as well. The map is keyed by the folder the item is in, with top-level
+     * items having an empty key.
+     */
+    public Map<String, Collection<TopLevelItem>> getItems() {
+        String recursive = Stapler.getCurrentRequest().getParameter("recursive");
+        if (recursive == null) {
+            return Collections.singletonMap("", view.getItems());
+        } else {
+            return Collections.unmodifiableMap(getItemsRecursive("", view.getItems()));
+        }
+    }
+
+    private Map<String, Collection<TopLevelItem>> getItemsRecursive(String namePrefix, Collection<TopLevelItem> items) {
+        Map<String, Collection<TopLevelItem>> result = new HashMap<>();
+        List<TopLevelItem> currentLevelItems = new ArrayList<>();
+        for (TopLevelItem i : items) {
+            if (i instanceof ItemGroup) {
+                ItemGroup g = (ItemGroup) i;
+                result.putAll(getItemsRecursive(namePrefix + g.getDisplayName() + "/", g.getItems()));
```

Comment:
I am not sure about `Display Name` here. It causes some concerns since you duplicate the core's logic here. I would rather suggest to use `Item#getFullDisplayName` though it decreases the performance of the call

Reply:
I didn't know about `Item#getFullDisplayName`, so I will update it to use that.